scipy/stats/_continuous_distns.py

```diff
@@ -4219,23 +4220,50 @@ def fit(self, data, *args, **kwds):
 laplace = laplace_gen(name='laplace')
 
 
-def _check_fit_input_parameters(data, args, kwds, fixed_param):
-    if len(args) > 0:
-        raise TypeError("Too many arguments.")
-
-    _remove_optimizer_parameters(kwds)
-
-    if None not in fixed_param:
+def _check_fit_input_parameters(self, data, args, kwds):
+    data = np.asarray(data)
+    floc = kwds.get('floc', None)
+    fscale = kwds.get('fscale', None)
+
+    num_shapes = len(self.shapes) if self.shapes else 0
+    fshape_keys = []
+    fshapes = []
+
+    # user has many options for fixing the shape, so here we standardize it
+    # into 'f' + the number of the shape.
```

Comment:
Maybe make a note of where this code was adapted from. The main reason is to tell the reviewer that this code may look funny, but we didn't want to fix it if it's not broken.

Reply:
It doesn't say this is outdated, but I added a comment for that on the next line